### PR TITLE
Add options <ignoredgetparams/> and <ignoredpostparams/>

### PR DESCRIPTION
**Pull request by balat**

### CODE REVIEW

File: src/lib/eliom_common.server.ml [hunk 1]
```diff
@@ -895,11 +895,20 @@ type cpi = client_process_info =  {
...
 } [@@deriving json]
 
-let get_session_info req previous_extension_err =
+let match_regexp name rex =
+  try
+    let _ = Pcre.exec ~rex ~flags:[`ANCHORED] ~pos:0 name in
+    true
+  with Not_found -> false
+
+let do_not_match_regexps regexps (name, _) =
```
Comment (jrochel) [inline src/lib/eliom_common.server.ml:904]: And this one `does_not_match_regexps`.

**Comment by vouillon** [comment]
Does is work when a link is opened in a app? I don't see any client-side filtering

**Comment by balat** [comment]
I don't really understand the change I did in eliom_registration.server.ml
What is this create_extension function?
The extension is already registered in eliommod.ml (and pagegen already called) ...

**Comment by balat** [comment]
I son't really understand the change I had to make in `eliom_registration.server.ml`.
This seems related to an `auto_load_extensions` feature.
Does anyone know what it is? Is it used somewhere? May be to create an standalone program without config file?
@vasilisp?

File: src/lib/eliom_process.client.ml [hunk 2]
```diff
@@ -50,6 +50,17 @@ let set_sitedata, is_set_sitedata,
     reset_sitedata =
   get_set_js_serverside_value Eliom_common.sitedata "__eliom_appl_sitedata"
 
+let ignored_get_params = ref []
+let ignored_post_params = ref []
+
+let set_ignored_params get post =
+  let compile =
+    List.map
+      (fun s -> print_endline s; Re.seq [Re.start; Re.Pcre.re s; Re.stop] |> Re.compile)
```
Comment (vouillon) [inline src/lib/eliom_process.client.ml:59]:
```suggestion
      (fun s -> Re.seq [Re.start; Re.Pcre.re s; Re.stop] |> Re.compile)
```

File: src/lib/eliom_registration.server.ml [hunk 3]
```diff
@@ -1584,7 +1586,8 @@ let extension =
        let sitedata = Eliommod.create_sitedata vh site_dir conf_info in
        Eliom_common.absolute_change_sitedata sitedata;
        (* CHECKME *)
-       Eliommod.preload ();
+       Eliom_common.begin_load_eliom_module ();
+       Eliommod.site_init (ref true);
```
Comment (jrochel) [inline src/lib/eliom_registration.server.ml:1590]: @vasilisp Does this make sense?

**Pull request merged**